etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataOrchestrator.java

```diff
@@ -71,19 +98,23 @@ public void updateMetadataForSentOrder(String receivedSubmissionId, String sentS
                     "Unable to retrieve metadata from RS history API", e);
         }
 
+        logger.logInfo("Updating metadata with receiver: {}", receiver);
         partnerMetadata = partnerMetadata.withReceiver(receiver);
         partnerMetadataStorage.saveMetadata(partnerMetadata);
     }
 
     public Optional<PartnerMetadata> getMetadata(String receivedSubmissionId)
             throws PartnerMetadataException {
-        // call the metadata storage to get the metadata.
-        // check if the receiver is filled out, and if it isn't, call the RS history API to get the
-        // receiver.
-        // if had to call the history API, extract the receiver and call the metadata storage to
-        // save the metadata with the receiver added.
-        // return the metadata.
-        return partnerMetadataStorage.readMetadata(receivedSubmissionId);
+        PartnerMetadata partnerMetadata =
+                partnerMetadataStorage.readMetadata(receivedSubmissionId).orElseThrow();
```

Comment:
The `.orElseThrow()` throws a `NoSuchElementException` which is an unchecked `RuntimeException`.  We should instead throw our `PartnerMetadataException`, right?  If that's right, then this would turn into...

```suggestion
                partnerMetadataStorage.readMetadata(receivedSubmissionId).orElseThrow(() -> new PartnerMetadataException("Can't update metadata for sent order, no metadata exists for ID " + receivedSubmissionId));
```

Reply:
Also, this should apply to the other `orElseThrow` in the `getMetadata` method.

Reply:
Good catch. I think this was a something that we thought about before the PartnerMetadataException was made, that we  needed a custom exception.

Reply:
I'll refactor `updateMetadataForSentOrder` to pass PartnerMetadata so we don't need to read the metadata again when we call it